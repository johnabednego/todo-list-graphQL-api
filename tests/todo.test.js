require('dotenv').config();
const { gql } = require('apollo-server-lambda');
const { createTestClient } = require('apollo-server-testing');
const { ApolloServer } = require('apollo-server-lambda');
const typeDefs = require('../src/schemas/typeDefs');
const resolvers = require('../src/resolvers/todoResolvers');
const connectDB = require('../src/utils/db');
const mongoose = require('mongoose');

jest.mock('../src/models/todo', () => ({
  find: jest.fn().mockResolvedValue([
    {
      id: '1',
      title: 'Test Todo 1',
      description: 'Description 1',
      completed: false,
    },
    {
      id: '2',
      title: 'Test Todo 2',
      description: 'Description 2',
      completed: false,
    },
  ]),
}));

beforeAll(async () => {
  await connectDB();
});

afterAll(async () => {
  await mongoose.connection.close();
});

describe('Queries', () => {
  it('fetches list of todos', async () => {
    const server = new ApolloServer({
      typeDefs,
      resolvers,
    });

    const { query } = createTestClient(server);
    const GET_TODOS = gql`
      query GetTodos {
        getTodos {
          id
          title
          description
          completed
        }
      }
    `;

    const response = await query({ query: GET_TODOS });
    expect(response).toMatchSnapshot();
    expect(response.data.getTodos).toHaveLength(2);
  });
});
