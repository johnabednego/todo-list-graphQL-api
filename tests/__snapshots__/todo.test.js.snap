// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries fetches list of todos 1`] = `
{
  "data": {
    "getTodos": [
      {
        "completed": false,
        "description": "Description 1",
        "id": "1",
        "title": "Test Todo 1",
      },
      {
        "completed": false,
        "description": "Description 2",
        "id": "2",
        "title": "Test Todo 2",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": {
    "headers": Headers {
      Symbol(map): {},
    },
  },
}
`;
